tensorboard/plugins/graph/graphs_plugin.py

```diff
@@ -55,17 +60,44 @@ def is_active(self):
 
   def index_impl(self):
     """Returns a list of all runs that have a graph."""
-    return [run_name
-            for (run_name, run_data) in self._multiplexer.Runs().items()
-            if run_data.get(event_accumulator.GRAPH)]
-
-  def run_metadata_index_impl(self):
-    """Returns a run-to-tag mapping for metadata."""
-    return {
-        run_name: run_data[event_accumulator.RUN_METADATA]
-        for (run_name, run_data) in self._multiplexer.Runs().items()
-        if event_accumulator.RUN_METADATA in run_data
-    }
+    result = {}
+    def add_row_item(run, tag=None):
+      run_item = result.setdefault(run, {
+          'run': run,
+          'tags': {},
+          'run_graph': False})
+
+      tag_item = None
+      if tag:
+        tag_item = run_item.get('tags').setdefault(tag, {
+            'tag': tag,
+            'conceptual_graph': False,
+            'op_graph': False,
+            'profile': False})
+      return (run_item, tag_item)
+
+    mapping = self._multiplexer.PluginRunToTagToContent(
+        _PLUGIN_NAME_RUN_METADATA_WITH_GRAPH)
+    for (run_name, tag_to_content) in six.iteritems(mapping):
+      for (tag, content) in six.iteritems(tag_to_content):
+        if content is not '1':
```

Comment:
Is this supposed to be a naked version number represented as a string?

Why isn’t this a serialized proto like all the other plugins use? As
written, it’s impossible to determine what’s setting the version to `1`.
If this were a named proto then I could just grep for the unique line of
non-test code that instantiates that proto.

Reply:
The point of the doc I tagged in the changelist you mentioned above is that proto for the 1. plugin content should not be defined in TensorFlow and 2. TensorFlow modules cannot depend on protos defined in TensorBoard. After some deliberation, we decided to use the plugin name as the key to differentiate types of data and have the minimal versioning scheme via the plugin content.